tools/object/types.go

```diff
@@ -0,0 +1,23 @@
+package object
+
+import (
+	"github.com/opencontainers/go-digest"
+)
+
+type RefEngine struct {
+	Protocol string `json:"protocol,omitempty"`
+	Uri      string `json:"uri,omitempty"`
```

Comment:
This template is specific to the `oci-index-template-v1` protocol.  You [can rely on `protocol`][1].  After that, you should hold the config in an `interface{}`.

[1]: https://github.com/xiekeyang/oci-discovery/blame/44ec3cf3113e29a743ad04220ccb7ff5197dab2a/ref-engine-discovery.md#L68